### PR TITLE
Zivid Python RC review

### PR DESCRIPTION
**Pull request by nedrebo**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+project(zivid LANGUAGES CXX)
+
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+if(MSVC)
+    add_compile_options(/bigobj)
+endif()
+
+find_package(Zivid COMPONENTS Core 1.3.0 REQUIRED)
```
Comment (StianP-Z) [inline CMakeLists.txt]: This doesn't work (I discovered the same). Problem 1 is that 1.3.0 needs to be moved to right of Zivid, before COMPONENTS. Right now it is a noop (and no error it seems).

 When it is moved to correct location, problem 2 is that it does an exact match, even when EXACT is not specified. 

So you get a

CMake Error at /home/stian/catkin_ws/src/zivid_ros/zivid_camera/CMakeLists.txt:33 (find_package):
  Could not find a configuration file for package "Zivid" that is compatible
  with requested version "1.2".

  The following configuration files were considered but not accepted:

    /usr/lib/cmake/Zivid/ZividConfig.cmake, version: 1.3.0

It looks like by looking at /usr/lib/cmake/Zivid/ZividConfig-version.cmake that we say compatible only for exact match, where we probably want to implement a greater than or equal check.

I temporarily worked around this in ROS branch by making my own check.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
+# Zivid Python
+
+
+Zivid Python is the official Python package for Zivid 3D cameras. Read more about Zivid at [zivid.com](https://www.zivid.com/).
```
Comment (StianP-Z) [inline README.md]: Btw I adjusted this part a bit in my PR. (Removed last part, added link for Zivid 3D cameras)

File: doc/scratchpad/sixtynorth-workshop/austin/hdr_examples.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
+from contextlib import contextmanager
```
Comment (StianP-Z) [inline doc/scratchpad/sixtynorth-workshop/austin/hdr_examples.py]: this stuff should probably not be included?

File: src/Common/include/ZividPython/Releasable.h [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    auto functionName()                                                                                                \
+    {                                                                                                                  \
+        return ReturnType{ impl().functionName() };                                                                    \
+    }
+
+#define ZIVID_PYTHON_RELEASABLE_FORWARD_0_ARGS_WRAP_CONTAINER_RETURN(Container, ReturnType, functionName)              \
+    auto functionName()                                                                                                \
+    {                                                                                                                  \
+        auto nativeContainer = impl().functionName();                                                                  \
+        Container<ReturnType> returnContainer;                                                                         \
```
Comment (StianP-Z) [inline src/Common/include/ZividPython/Releasable.h]: candidate for reserve()

File: src/Core/HDR.cpp [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+namespace py = pybind11;
+
+namespace ZividPython::HDR
+{
+    MetaData wrapAsSubmodule(pybind11::module &dest)
+    {
+        dest.def(
+            "combine_frames",
+            [](const std::vector<ReleasableFrame> &releasableFrames) {
+                std::vector<Zivid::Frame> frames;
```
Comment (StianP-Z) [inline src/Core/HDR.cpp]: candidate for reserve

File: src/Core/PointCloud.cpp [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+        static_assert(sizeof(NativeDataType) == sizeof(DataType), "Unexpected point cloud format");
+        static_assert(IS_SAME_MEMBER(NativeDataType, DataType, x), "Unexpected point cloud format");
+        static_assert(IS_SAME_MEMBER(NativeDataType, DataType, y), "Unexpected point cloud format");
+        static_assert(IS_SAME_MEMBER(NativeDataType, DataType, z), "Unexpected point cloud format");
+        static_assert(IS_SAME_MEMBER(NativeDataType, DataType, contrast), "Unexpected point cloud format");
+        static_assert(offsetof(NativeDataType, rgba) == offsetof(DataType, r), "Unexpected point cloud format");
+        static_assert(sizeof(NativeDataType::rgba)
+                          == sizeof(DataType::r) + sizeof(DataType::g) + sizeof(DataType::b) + sizeof(DataType::a),
+                      "Unexpected point cloud format");
```
Comment (StianP-Z) [inline src/Core/PointCloud.cpp]: nice stuff

File: test/test_environment.py [hunk 7]
```diff
@@ -0,0 +1,9 @@
+def test_data_path(application):  # pylint: disable=unused-argument
+    import zivid
+    import pathlib
+    import tempfile
+    import os
+
+    with tempfile.TemporaryDirectory() as temp_dir:
+        os.environ["ZIVID_DATA"] = temp_dir
+        assert isinstance(zivid.environment.data_path(), pathlib.Path)
```
Comment (StianP-Z) [inline test/test_environment.py]: here is missing check that path is temp_dir?

File: CMakeLists.txt (hunk 1 above)
Comment (nedrebo) [reply CMakeLists.txt]: Did you create a SDK issue for this?

File: doc/scratchpad/sixtynorth-workshop/austin/hdr_examples.py (hunk 3 above)
Comment (nedrebo) [reply doc/scratchpad/sixtynorth-workshop/austin/hdr_examples.py]: No, deleted in fixup branch

File: src/Common/include/ZividPython/Releasable.h (hunk 4 above)
Comment (nedrebo) [reply src/Common/include/ZividPython/Releasable.h]: Fixed

File: README.md (hunk 2 above)
Comment (nedrebo) [reply README.md]: Ok, ill leave it as is

File: src/Core/PointCloud.cpp (hunk 6 above)
Comment (nedrebo) [reply src/Core/PointCloud.cpp]: thx

File: src/Core/HDR.cpp (hunk 5 above)
Comment (nedrebo) [reply src/Core/HDR.cpp]: fixed

File: test/test_environment.py (hunk 7 above)
Comment (nedrebo) [reply test/test_environment.py]: added

File: CMakeLists.txt (hunk 1 above)
Comment (StianP-Z) [reply CMakeLists.txt]: I found out there is an existing bug for this, ZIVID-1409

**Pull request closed without merging**